### DISCUSSION TITLE
supabase.auth.user() is returning user as null

### DISCUSSION THREAD

**Discussion by nickbyte** [supabase/supabase#3222] [Questions] [answered]
Using the new phone auth, when trying to fetch the user id, the user ID is null, how to resolve this?

**Comment by notjustinshaw** [comment]
I have the same problem. Signing in works fine but the response from verifyOTP returns a `null` user object even though the session is valid.

**Comment by kangmingtay** [comment, marked as answer]
Hey @nickbyte and @notjustinshaw, 

Thanks for pointing this out! You can obtain the user information from the access token JWT payload. The user ID can be retrieved from the "sub" field in the JWT as it's the same as the "id" field stored in the database. 

Hope this helps!

**Reply by CaseyG5** [reply]
Hi @kangmingtay, 

Would you care to share more about the web token and how to pull the user ID from it?  For me, both `user` and `session` that `supabase.auth.signUp()` supposedly returns are undefined and null, respectively.

**Reply by A7DC** [reply]
Same, did you find a solution @CaseyG5 ?
